src/tsd/GraphHandler.java

```diff
@@ -69,20 +70,21 @@
     = new AtomicInteger();
 
   /** Keep track of the latency of graphing requests. */
-  private static final Histogram graphlatency =
-    new Histogram(16000, (short) 2, 100);
+  private final LatencyStatsPlugin graphlatency;
```

Comment:
Actually, leave all these static. Use your ``getInstance();`` calls to instantiate the plugins. Then you don't have to modify all of the collection code.